JavaScript/Angular.md

```diff
@@ -0,0 +1,687 @@
+
+*Guides are not rules and should not be followed blindly. Use your head and think.*
+
+## Core Libraries and Tools
+
+### [Angular CLI](https://github.com/angular/angular-cli)
+
+If you are starting a new project, the best way to create a new project is by using Angular CLI. It will hide a lot of configuration behind the scenes (you will not have a webpack config exposed for editing), it does some optimizations when compiling your code and it offers some handy scaffolding tools.
+
+Install Angular CLI globally with `npm install -g @angular/cli` and check out what you can do with `ng help`.
+
+#### Creating a new project
+
+Use Angular CLI for initial project setup:
+
+```bash
+ng new my-app
+```
+
+As of version 7 there will be a quick wizard-like setup prompting you about routing and styling. Choose with routing and SCSS styling.
+
+#### Creating new components
+
+Create component with `ng g c MyComponent` or `ng g c my-component`. In both cases resulting component class name will be `MyComponent` and dir name will be `my-component`.
+
+#### Creating new services
+
+Use `ng g s MyService` to generate new service. NOTE: unlike during component generation, service generation will not create a new dir for the service (although that might change in the future).
+
+#### Ejecting
+
+Do not do it, but if you **ABSOLUTELY HAVE TO** edit the webpack config manually, you can eject the application with `ng eject`. Unfortunately there is no way to extend webpack config.
+
+#### Devserver proxy
+
+If you have issues with CORS, for development purpuses it is OK to temporarily use Webpack Devserver proxy. You can do this without ejecting the webpack config, [as per instructions](https://angular.io/guide/build#proxying-to-a-backend-server).
+
+#### Other commands
+
+This handbook will not cover all `ng` commands, please check out the [wiki](https://github.com/angular/angular-cli/wiki).
+
+### Typings
+
+If you are using some JS library which is now written in TypeScript, you can usually install typings separately. Install typings as dev-dependency, for example:
+
+```bash
+npm install --save d3
+npm install --save-dev @types/d3
+```
+
+### [RxJS](https://github.com/ReactiveX/rxjs)
+
+Angular depends heavily on RxJS so getting to know RxJS is pretty important. It is used for all event handling and can even be used for application state management of some description.
+
+Please check out some RxJS tutorials if you are not familiar with Rx.
+
+#### Observers and Subscribers
+
+When you want to observe some value, use some of the classes derived from `Observable`. Most use cases are coverd with one of these:
+
+- `Subject`
+- `BehaviorSubject`
+- `ReplaySubject`
+
+All of classed derived from `Observable` can be subscribed to in order to receive value changes. You can also use `operators` to manipulate how and when the value changes are sent to the subscribers.
+
+One convention is that variables which are `Observable` are postfixed with a dollar sign, like so:
```

Comment:
If this is something that we practice as well, maybe the formulation should be more of a declarative one: "`Observable` variables should be prefixed with a dollar sign ...".